src/Bridge/Doctrine/MongoDbOdm/Filter/SearchFilter.php

```diff
@@ -166,20 +166,24 @@ private function addEqualityMatchStrategy(string $strategy, string $field, $valu
     {
         $type = $metadata->getTypeOfField($field);
 
+        if (MongoDbType::STRING !== $type) {
+            return MongoDbType::getType($type)->convertToDatabaseValue($value);
+        }
```

Comment:
Why this check? Did you encounter another issue?

Reply:
I just moved it upwards, away from the `switch` below.